source/common/common/utility.cc

```diff
@@ -248,14 +249,27 @@ std::string StringUtil::escape(const std::string& source) {
 }
 
 std::string AccessLogDateTimeFormatter::fromTime(const SystemTime& time) {
-  static DateFormatter date_format("%Y-%m-%dT%H:%M:%S");
+  static DateFormatter default_date_format("%Y-%m-%dT%H:%M:%S");
 
   return fmt::format(
-      "{}.{:03d}Z", date_format.fromTime(time),
+      "{}.{:03d}Z", default_date_format.fromTime(time),
       std::chrono::duration_cast<std::chrono::milliseconds>(time.time_since_epoch()).count() %
           1000);
 }
 
+std::string AccessLogDateTimeFormatter::fromTimeWithFormat(const SystemTime& time,
+                                                           const std::string& format) {
+
+  if (format != "") {
+    time_t in_time_t = std::chrono::system_clock::to_time_t(time);
+    std::stringstream ss;
+    ss << std::put_time(std::localtime(&in_time_t), format.c_str());
```

Comment:
There is an existing helper class to do this formatting already in source/common/common/utility.h: DateFromatter. Please use that.